### PR TITLE
Added dump persistence schema example app

### LINKED ISSUE #216: Add real dumping of persistence-schema to `examples/`
As for now, compiling the example application: `examples/akka-cluster-app` with `dump-persistence-schema-plugin` enabled does not produce any dump. We should add a working example for dumping persistence schema.

Still to be decided, which option to choose:
a) Add persistence schema to `examples/akka-cluster-app` (modify app's logic)
b) Leave `examples/akka-cluster-app` unchanged and add another example app under `examples` - just for this need

### PR DESCRIPTION
**Pull request by LukaszKontowski**
Closes #216

### CODE REVIEW

File: .github/workflows/standard-workflow.yml [hunk 1]
```diff
@@ -41,6 +41,10 @@ jobs:
         run: sbt scalafmtCheckAll scalafmtSbtCheck "scalafixAll --check"
         working-directory: examples/akka-cluster-app
 
+      - name: Check code style - examples/akka-persistence-app
+        run: sbt scalafmtCheckAll scalafmtSbtCheck "scalafixAll --check"
```
Comment (PawelLipski) [inline .github/workflows/standard-workflow.yml:45]: Actually... you can use YAML anchor here, see https://github.com/VirtusLab/git-machete-intellij-plugin/blob/develop/.circleci/config.yml#L222

File: pre-commit [hunk 2]
```diff
@@ -11,3 +11,6 @@ sbt "scalafmtAll" "scalafixAll" "scalafmtSbt"
 
 cd ../akka-cluster-app
 sbt "scalafmtAll" "scalafixAll" "scalafmtSbt"
+
+cd ../akka-persistence-app
+sbt "scalafmtAll" "scalafixAll" "scalafmtSbt"
```
Comment (PawelLipski) [inline pre-commit:16]: Nit:
```suggestion
sbt scalafmtAll scalafixAll scalafmtSbt
```

And in other lines. Shell quoting is redundant here

File: examples/akka-persistence-app/src/main/scala/org.virtuslab/example/ShoppingCart.scala [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        case _ => copy(items = items + (itemId -> quantity))
+      }
+    }
+  }
+  object State {
+    val empty: State = State(items = Map.empty)
+  }
+
+  // codecs derivation for all types extending CirceAkkaSerializable
+  implicit lazy val codecActorRefResponse: Codec[ActorRef[StatusReply[Summary]]] = new AkkaCodecs {}.actorRefCodec
```
Comment (PawelLipski) [inline examples/akka-persistence-app/src/main/scala/org.virtuslab/example/ShoppingCart.scala:97]: Huh while you're here... maybe you could add `object AkkaCodecs extends AkkaCodecs`? coz `new ... {}` looks awkward as hell

File: .github/workflows/standard-workflow.yml (hunk 1 above)
Comment (LukaszKontowski) [reply .github/workflows/standard-workflow.yml:45]: Are you sure this is not specific for circleci?

File: examples/akka-persistence-app/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    sbt "run local2.conf"
+    ```
+
+5. (Optional) Check for service readiness
+
+    ```
+    curl http://localhost:9101/ready
+    ```
+
+6. Interact with the service using `grpcurl` - example request (put items in the cart):
```
Comment (PawelLipski) [inline examples/akka-persistence-app/README.md:42]: Won't harm to link to https://github.com/fullstorydev/grpcurl

File: examples/akka-persistence-app/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+## Dump Persistence Schema Compiler Plugin usage
+You can test Dump Persistence Schema Compiler Plugin on this example project. There are two possibilities to do it:
+- `sbt compile` - compiles the project with the `dump-persistence-schema` phase enabled. `dump-persistence-schema` phase dumps detected persistence schema (types for Events and States) into `.json` files under the `target/dump-psersistence-schema-cache` directory of this project. These `.json` files are cache files used later by the `ashDumpPersistenceSchema` sbt task.
+- `sbt ashDumpPersistenceSchema` &mdash; sbt task that creates final output of the dump using mentioned `.json` files. Output is saved into the `target/akka-persistence-app-dump-persistence-schema-0.1.0-SNAPSHOT.yaml` file.
+You can just invoke `sbt ashDumpPersistenceSchema` without `sbt compile` &mdash; `ashDumpPersistenceSchema` task will invoke `sbt compile` if needed.
+
+## Running the sample code
+
+1. Prepare docker containers with:
```
Comment (PawelLipski) [inline examples/akka-persistence-app/README.md:12]: Okay... this entire setup is yours? or has it been taken from the original example?

File: .github/workflows/standard-workflow.yml (hunk 1 above)
Comment (LukaszKontowski) [reply .github/workflows/standard-workflow.yml:45]: https://github.com/actions/runner/issues/1182 - looks like YAML anchors are not available for .github workflows yet?

File: examples/akka-persistence-app/README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+    ```
+    curl http://localhost:9101/ready
+    ```
+
+6. Interact with the service using `grpcurl` - example request (put items in the cart):
+
+    ```
+    grpcurl -d '{"cartId":"cart1", "itemId":"T-shirt", "quantity":3}' -plaintext 127.0.0.1:8101 shoppingcart.ShoppingCartService.AddItem
+    ```
```
Comment (PawelLipski) [inline examples/akka-persistence-app/README.md:46]: I'd actually add an invocation of `dumpPersistenceSchema`, and point to the generated dump file.

**Comment by LukaszKontowski** [comment]
object `AkkaCodecs` etending the `AkkaCodecs` trait added. However - to change usage in examples, new release of ASH would be needed.

File: examples/akka-persistence-app/README.md [hunk 7]
```diff
@@ -33,14 +33,21 @@ You can just invoke `sbt ashDumpPersistenceSchema` without `sbt compile` &mdash;
...
+6. Interact with the service using [grpcurl](https://github.com/fullstorydev/grpcurl) - example request (put items in the cart):
 
     ```
     grpcurl -d '{"cartId":"cart1", "itemId":"T-shirt", "quantity":3}' -plaintext 127.0.0.1:8101 shoppingcart.ShoppingCartService.AddItem
     ```
+
+7. Dump persistence schema with Akka Serialization Helper (after shutting down the application):
+
+    ```
+    `sbt ashDumpPersistenceSchema`
```
Comment (PawelLipski) [inline examples/akka-persistence-app/README.md:51]:
```suggestion
    sbt ashDumpPersistenceSchema
```

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**